.github/workflows/reusable_test_suite.yml

```diff
@@ -23,30 +23,41 @@ jobs:
     name: 'Test suite'
     runs-on: ubuntu-latest
     container:
-      image: ghcr.io/mesh-adaptation/${{ inputs.docker-image }}:latest
+      image: ghcr.io/mesh-adaptation/${{ inputs.docker-image }}:pip
       options: --user root
+    env:
+      # Allow executing mpiexec as root
+      OMPI_ALLOW_RUN_AS_ROOT: '1'
+      OMPI_ALLOW_RUN_AS_ROOT_CONFIRM: '1'
+      # Since we are root we need to set PYTHONPATH to find the installed packages
+      PYTHONPATH: /home/firedrake/firedrake:/home/firedrake/animate:/home/firedrake/goalie:/home/firedrake/movement:/home/firedrake/.local/lib/python3.12/site-packages
 
     steps:
       - name: 'Check out the repo'
         id: checkout
-        uses: actions/checkout@v4     
+        uses: actions/checkout@v4
 
       - name: 'Install package'
-        run: |
-          . /home/firedrake/firedrake/bin/activate
-          git config --global --add safe.directory '*'
-          make install_dev
+        run: pip install --break-system-packages -e .[dev]
 
       - name: 'Lint'
+        run: make lint
+
+      - name: 'Run serial tests'
+        run: |
+          python3 $(which firedrake-clean)
+          python3 -m coverage erase
+          python3 -m coverage run --parallel-mode --source=${{ github.event.repository.name }} -m pytest -v -k "parallel[1] or not parallel" test
+
+      - name: 'Run parallel tests (nprocs = 2)'
         run: |
-          . /home/firedrake/firedrake/bin/activate
-          make lint
+          if pytest --collect-only -m parallel[2] test | grep -q 'no tests collected'; then
+            echo "No parallel[2] tests found."
+          else
+            mpiexec -n 2 python3 -m coverage run --parallel-mode --source=${{ github.event.repository.name }} -m pytest -v -m parallel[2] test
+          fi
```

Comment:
This is neat, thanks. Do we not need to account for np=3, too?

Reply:
Nevermind, I just saw your comment on https://github.com/mesh-adaptation/animate/pull/175.